lib/types.js

```diff
@@ -2111,6 +2116,12 @@ RecordType.prototype._createConstructor = function () {
       ds.push(defaultValue);
     }
   }
+  // istanbul ignore else
+  if (this._isError && this._showStack) {
+    innerBody += 'if (!this.stack) {\n';
```

Comment:
The check should be `this.stack === undefined`. (Let's also indent the statements by two extra spaces to align them with the rest. I'd also just put them all one a single line, similar to the others above, and add a new line at the very end.)